src/components/Sidebar/PostDetails/NewCommentForm.tsx

```diff
@@ -0,0 +1,241 @@
+import React, { useContext, useState } from 'react';
+import classNames from 'classnames';
+import { addComment } from '../../../utils/fetchFunctions';
+import {
+  ActivePostContext,
+  CommentListContext,
+  ErrorsContext,
+} from '../../../utils/Store';
+import { ErrorTypes } from '../../../types/ErrorTypes';
+
+enum Keys {
+  nameField = 'nameField',
+  emailField = 'emailField',
+  textArea = 'textArea',
+}
+
+export const NewCommentForm: React.FC = () => {
+  const [isErrorField, setIsErrorField] = useState({
+    nameField: false,
+    emailField: false,
+    textArea: false,
+  });
+
+  const [inputQueries, setInputQueries] = useState({
+    nameField: '',
+    emailField: '',
+    textArea: '',
+  });
+
+  const [isProcessing, setIsProcessing] = useState(false);
+
+  const { activePost } = useContext(ActivePostContext);
+  const { comments, setComments } = useContext(CommentListContext);
+  const { isError, setIsError } = useContext(ErrorsContext);
+
+  const onChangeHandler = (property: Keys, value: string) => {
+    if (isErrorField[property]) {
+      setIsErrorField({
+        ...isErrorField,
+        [property]: false,
+      });
+    }
+
+    setInputQueries({ ...inputQueries, [property]: value });
+  };
+
+  function handleClear() {
+    setInputQueries({
+      nameField: '',
+      emailField: '',
+      textArea: '',
+    });
+
+    setIsErrorField({
+      nameField: false,
+      emailField: false,
+      textArea: false,
+    });
+  }
+
+  function handleAdd(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault();
+
+    const newError = { ...isErrorField };
+
+    Object.values(Keys).forEach(key => {
+      if (!inputQueries[key].trim()) {
+        newError[key] = true;
+      }
+    });
```

Comment:
Ensure that you cannot add a movie with empty data (with spaces only). The current implementation checks for trimmed input, which is correct, but make sure this logic is consistently applied across all fields.